os_migrate/roles/import_workloads/tasks/workload.yml

```diff
@@ -21,6 +21,17 @@
         - "{{ prelim.server_name }} progress file: {{ prelim.state_file }}"
     when: prelim.changed
 
+  - name: perform required checks to ensure source workload is ready
+    os_migrate.os_migrate.import_workload_source_check:
+      auth: "{{ os_migrate_src_auth }}"
+      auth_type: "{{ os_migrate_src_auth_type|default(omit) }}"
+      region_name: "{{ os_migrate_src_region_name|default(omit) }}"
+      validate_certs: "{{ os_migrate_src_validate_certs|default(omit) }}"
+      ca_cert: "{{ os_migrate_src_ca_cert|default(omit) }}"
+      client_cert: "{{ os_migrate_src_client_cert|default(omit) }}"
+      client_key: "{{ os_migrate_src_client_key|default(omit) }}"
+      name: "{{ prelim.server_name }}"
```

Comment:
We should be looking up by ID, all the workload migration modules use IDs for the source server, the server_name is just for display purposes and logs. E.g. https://github.com/os-migrate/os-migrate/blob/f3d6dd70ad65c9ce403e76a76b4c2f18714423ab/os_migrate/plugins/modules/import_workload_export_volumes.py#L507 .

We have some options:

* use `name: "{{ item['_info']['id'] }}"` here, no other changes necessary i think,

* use `name: "{{ prelim.src_server_id }}"` here and add `result['src_server_id'] = info['id']` into the prelim module: https://github.com/os-migrate/os-migrate/blob/f3d6dd70ad65c9ce403e76a76b4c2f18714423ab/os_migrate/plugins/modules/import_workload_export_volumes.py#L507. This might be nicer as when running Ansible with -v, we get this ID printed out into the logs too.

Reply:
(I edited the above comment - one of the source links was wrong, should be good now.)

Reply:
I noticed one more thing, we need to give this `when: prelim.changed` in case the prelim module decided to skip the migration because it was trying to migrate the conversion host itself.